modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp

```diff
@@ -1061,16 +1061,15 @@ class WaveInputDeviceInstance  : public InputDeviceInstance
         // BEATCONNECT MODIFICATION START (RELAY)
         if (wi.getIsRelay() && wi.isEnabled())
         {
-            // Reset the size to 0.
-            // realyBufferProcessor will set the correct size.
-            inputBuffer.setSize(inputBuffer.getNumChannels(), 0, false, true);
-
             // relayBufferProcessor is a shared resource.
             // This callback function will be set to nullptr when the Relay disconnects or the application is closed.
             std::unique_lock<std::mutex> lock(edit.engine.getDeviceManager().m_muRelayCallback);
 
             if (nullptr != edit.engine.getDeviceManager().relayBufferProcessor)
-                edit.engine.getDeviceManager().relayBufferProcessor(inputBuffer);
+            {
+                bool isReady = (recordingContext != nullptr && recordingContext->adjustSamples < numSamples);
```

Comment:
I recommend considering the case where the adjustSamples is not a multiple of numSamples. In this case, there will be an offset.

Reply:
isReady is used to know when to start retrieving from the relay.
This is the same condition as for the start of the recording.